packages/thirdweb/src/react/web/ui/prebuilt/thirdweb/BuyButton/index.tsx

```diff
@@ -0,0 +1,186 @@
+"use client";
+
+import type { Chain } from "../../../../../../chains/types.js";
+import type { ThirdwebClient } from "../../../../../../client/client.js";
+import { getContract } from "../../../../../../contract/contract.js";
+import { getListing } from "../../../../../../extensions/marketplace/direct-listings/read/getListing.js";
+import type { BaseTransactionOptions } from "../../../../../../transaction/types.js";
+import { useReadContract } from "../../../../../core/hooks/contract/useReadContract.js";
+import type { TransactionButtonProps } from "../../../../../core/hooks/transaction/transaction-button-utils.js";
+import { useSendAndConfirmTransaction } from "../../../../../core/hooks/transaction/useSendAndConfirmTransaction.js";
+import { useActiveAccount } from "../../../../../core/hooks/wallets/useActiveAccount.js";
+import { TransactionButton } from "../../../TransactionButton/index.js";
+
+export type BuyButtonProps = Omit<TransactionButtonProps, "transaction"> & {
+  /**
+   * The contract address of the Marketplace v3 contract.
+   */
+  contractAddress: string;
+
+  /**
+   * The chain which the Drop contract is deployed on
+   */
+  chain: Chain;
+
+  /**
+   * thirdweb Client
+   */
+  client: ThirdwebClient;
+
+  /**
+   * ID of the marketplace's DirectListing
+   */
+  listingId: bigint;
```

Comment:
if this is for direct listings only then maybe we can name this `BuyDirectListingButton` or something? We'll likely also want a component later for auctions etc and I wonder if it would get confusing otherwise? (my name suggestion might also not be good but more explicit naming is probably beneficial here?)

Reply:
No I agree as that thought has been in my head. I don't want any vague name but I was also scared of naming things too "verbosely".

Now that you have mentioned, I'm gonna rename it.